### ISSUE TITLE
[no-magic-numbers] ignoreReadonlyClassProperties has no effect on bigint

### ISSUE THREAD

**Issue by pikadun** [typescript-eslint/typescript-eslint#4389]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "no-magic-numbers": "off",
    "@typescript-eslint/no-magic-numbers": ["warn", {
        "ignoreReadonlyClassProperties": true
    }]
  }
}
```

```TS
// your repro code case
class Test {
     private readonly START_MILLISECOND = 1600000000000n;
}
```

**Expected Result**

**Actual Result**

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `5.8.1` |
| `@typescript-eslint/parser`        | `5.8.1` |
| `TypeScript`                       | `4.5.4` |
| `ESLint`                           | `8.4.1` |
| `node`                             | `16.13.1` |

**Comment by bradzacher** [comment]
This check fails on bigints for obvious reasons
https://github.com/typescript-eslint/typescript-eslint/blob/8d710a096880e13473b74b4ae6e784dcc545d375/packages/eslint-plugin/src/rules/no-magic-numbers.ts#L75
